Registration/RemoteManagement.psm1

```diff
@@ -0,0 +1,581 @@
+        $VerbosePreference = [System.Management.Automation.ActionPreference]::Continue
+    
+        $enableRemoteManagementActionName = "enableRemoteManagement"
+        try
+        {
+            $registrationResouce = Get-AzureRmResource -ResourceId $RegistrationResourceId
+            if($null -eq $registrationResouce)
+            {
+                Log-Throw "Unable to find registration resource $registrationResouce"
+            }
+    
+            Log-Output "Found registration resource:$RegistrationResourceId. Now calling action:$enableRemoteManagementActionName on Registration:$($registrationResouce.Name) with resourceGroup:$($registrationResouce.resourceGroupName)"
+            Invoke-AzureRmResourceAction `
+                -ResourceName $registrationResouce.Name `
+                -ResourceGroupName $registrationResouce.resourceGroupName `
+                -Action $enableRemoteManagementActionName `
+                -ResourceType $registrationResouce.ResourceType `
+                -Force
+    
+            Log-Output "[Success]::Action:$enableRemoteManagementActionName on Registration Resource:$resourceName with resourceGroup:$resourceGroupName"
+        }
+        Catch
+        {
+            Log-Throw "An error occurred while calling action $enableRemoteManagementActionName on RegistrationResource:$RegistrationResourceId `r`n$($_)" -CallingFunction $PSCmdlet.MyInvocation.MyCommand.Name
+        }
+    }
+    
+    <#
+    .SYNOPSIS
+    Uses the current session with the PrivilegedEndpoint to get registration information need to create EdgeSubscription
+    #>
+    function Get-InfoForEdgeSubscription{
+    [CmdletBinding()]
+        Param(
+            [Parameter(Mandatory=$true)]
+            [System.Management.Automation.Runspaces.PSSession] $PSSession
+        )
+        
+        $ErrorActionPreference = [System.Management.Automation.ActionPreference]::Stop
+        $VerbosePreference = [System.Management.Automation.ActionPreference]::Continue
+    
+        try
+        {
+            Log-Output "Calling Get-DeviceInfoForEdgeSubscription from PEP endpoint."
+            $deviceInfo = Invoke-Command -Session $PSSession -ScriptBlock { Get-DeviceInfoForEdgeSubscription -WarningAction SilentlyContinue }
+        }
+        Catch
+        {
+            Log-Throw "An error occurred querying device and registration information to create linked subscription : `r`n$($_)" -CallingFunction $PSCmdlet.MyInvocation.MyCommand.Name
+        }
+    
+        Log-Output -Message "Received information $($deviceInfo | ConvertTo-Json -Depth 4)."
+        Log-Output -Message "Count: $($deviceInfo.Count)"
+        Log-Output -Message "DeviceInfo1: $($deviceInfo[1] | ConvertTo-Json -Depth 4)"
+    
+        return $deviceInfo[1]
+    }
+    
+    <#
+    .SYNOPSIS
+    Uses the current session with the PrivilegedEndpoint to enable remote management on device
+    #>
+    function Enable-RemoteManageOnDevice{
```

Comment:
Do we need a Disable- cmdlet?

Reply:
not right now. Disable is done directly from PEP. Documentation will call it out.